recipes/trantor/all/conanfile.py

```diff
@@ -52,59 +48,69 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("openssl/1.1.1q")
         if self.options.with_c_ares:
             self.requires("c-ares/1.18.1")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "14")
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
 
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
         if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("trantor requires C++14, which your compiler does not support.")
+            if Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
         else:
-            self.output.warn("trantor requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+            self.output.warn(f"{self.ref} requires C++{self._minimum_cpp_standard}. Your compiler is unknown. Assuming it supports C++{self._minimum_cpp_standard}.")
 
-        # TODO: Compilation succeeds, but execution of test_package fails on Visual Studio 16 MDd
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) == "16" and \
-           self.options.shared == True and self.settings.compiler.runtime == "MDd":
-            raise ConanInvalidConfiguration("trantor does not support the MDd runtime on Visual Studio 16.")
+        # TODO: Compilation succeeds, but execution of test_package fails on Visual Studio with MDd
+        if is_msvc(self) and self.options.shared and self.info.settings.compiler.runtime == "MDd":
```

Comment:
Is it possible to use [conan.tools.microsoft.is_msvc_static_runtime()](https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc-static-runtime) here?

Reply:
@jwillikers 
It was occured only on MDd runtime, not MD.
I think `is_msvc_static_runtime()` is too wide for it.
Is there any way to specify MDd only using `is_msvc_static_runtime()`?

Reply:
Right you are... ironically I just realized what we're probably looking for is only one item down: [conan.tools.microsoft.msvc_runtime_flag()](https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-msvc-runtime-flag). You should be able to check if that returns `MDd`. Using that method will work for both the `Visual Studio` compiler as well as the `msvc` compiler, where the value of `compiler.runtime` is either `static` or `dynamic`.

Reply:
@jwillikers 
Thank you for your explanation!
Now I understand the reason using msvc_runtime_flag.